### ISSUE TITLE
Ensure that non renderer systems compile on CL implementations

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2662]
All of our non-renderer specific code should be portable across CL implementions. 

With that in mind, I'm leaving this reminder for the next release.

Systems include:

- `nyxt`
- `nyxt/prompter`
- `nyxt/history-tree`
- etc

The implementations we should test with include (besides SBCL):

- CCL
- AllegroCL 
- Clisp
- ECL
- ABCL

**Comment by Ambrevar** [comment]
Actually both `nyxt/prompter` and `nyxt/history-tree` should be extracted before 3.0 ;)

**Comment by aadcg** [comment]
Totally agree. `prompter` only after #2655.

**Comment by Ambrevar** [comment]
Well, there are many prompter prompter issues...

https://github.com/atlas-engineer/nyxt/issues?q=is%3Aissue+is%3Aopen+label%3Aprompt-buffer

**Comment by aadcg** [comment]
There are only 2 issues that are both labelled as 3.0 and prompt buffer. The remaining prompt buffer issues should be moved to its new github repository.

**Comment by Ambrevar** [comment]
Yup, #2134 is very important and requires the major refactoring of switching from calispel to lparallel.

In any case, all these issues can be done after the move to a separate repo, it's mostly that it'll be easier to close as many as possible now and not later, to avoid having to bump versions (Guix update, submodule update, etc.).

**Comment by aadcg** [comment]
Yes, that particular issue is perhaps the most important one.

Also, perhaps not all of these prompt buffer issues are related to the library, but to prompt buffer and prompt buffer mode in the context of the `nyxt` system.

**Comment by Ambrevar** [comment]
Absolutely.

**Comment by aadcg** [comment]
This is an important goal, but it seems overkill to test every commit on all CL implementations. For each commit we test on SBCL and CCL (via the GitHub CI), which is already quite good.

Maybe testing on more CL implementation should be a practice done exclusively before releasing major versions.

**Comment by Ambrevar** [comment]
By the way `prompter` has been extract, it will soon build on all Lisps and #2134 is in the process of being fixed.
